test/rules/unnecessary_raw_strings_test.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(UnnecessaryRawStringsTest);
+  });
+}
+
+@reflectiveTest
+class UnnecessaryRawStringsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'unnecessary_raw_strings';
+
+  test_doubleQuotes_raw() async {
```

Comment:
I'd remove the '_raw' from every test method name because they're all raw strings (unless we think we should add tests of non-raw strings). But completely up to you.

Reply:
Yeah the non-raw case should be tested.

I've added a test case, and left the other names the same. Thanks for the catch!